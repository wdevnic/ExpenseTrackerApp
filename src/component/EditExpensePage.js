import React from 'react'
import {connect} from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { editExpense, removeExpense } from '../actions/expenses'


export  class EditExpensePage extends React.Component{
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense)
        this.props.history.push('/')
    }

    onClick=() =>{
        this.props.removeExpense({id: this.props.expense.id})
        this.props.history.push('/')
    }

    render() {
        return (
            <div>
                <ExpenseForm
                expense={this.props.expense}
                onSubmit = {this.onSubmit}> 
                </ExpenseForm>
    
                <button onClick={this.onClick}
                >Remove</button> 
            </div>
         )
    }
    
}

// props not needed here just for example
const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (data) => dispatch(removeExpense(data))
})

const mapStateToProps = (state, props) => {
    return{
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id
        })
    }
}

 export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)