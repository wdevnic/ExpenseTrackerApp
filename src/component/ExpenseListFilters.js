import React from 'react'
import { connect } from 'react-redux'
import  {DateRangePicker} from 'react-dates'
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters'


export class ExpenseListFilters extends React.Component {

    state = {
        calenderFocused: null
    }

    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }

    onFocusChange = (calenderFocused)=> {
        this.setState(() => ({calenderFocused}))
    }

    onSortChange = (e)=>{
        const selected = e.target.value
        if(selected === 'date'){
            this.props.sortByDate(selected)
        }
        else if(selected === 'amount'){
            this.props.sortByAmount(selected)
        }
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }


    render(){
        return (

            <div>
                <input 
                type="text" 
                value={this.props.filters.text} 
                onChange={this.onTextChange} />
        
                <select 
                value={this.props.filters.sortBy} 
                onChange={ this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker 
                startDate={this.props.filters.startDate}
                startDateId= {'startId'}
                endDateId ={'endId'}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calenderFocused}
                onFocusChange={this.onFocusChange}
                showClearDates= {true}
                numberOfMonths={1}
                isOutsideRange= {() => false}
                />
            
                
            </div>
        )
    }

}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})


const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)