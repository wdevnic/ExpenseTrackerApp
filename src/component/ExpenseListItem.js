import React from 'react'
import {Link} from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <div>
        <Link to={"/edit/" + id} activeClassName="is-active">
            <h3>{description}</h3>
        </Link>
        <p>{amount} - {createdAt}</p>
       
    </div>
)


export default ExpenseListItem

