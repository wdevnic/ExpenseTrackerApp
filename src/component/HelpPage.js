import React from 'react'

const HelpPage = () => (
    <div>
        This is help
    </div>
 )

 export default HelpPage