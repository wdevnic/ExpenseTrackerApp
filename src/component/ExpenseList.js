import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import getVisibleExpenses from "../selectors/expenses";

export const ExpenseList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expense</div>
      <div className="show-for-desktop">Amount</div>
    </div>
    <div className="list-body">
      {props.expenses.length === 0 ? (
        <div>
          <span className="list-item list-item--message">No expenses</span>
        </div>
      ) : (
        <ul>
          {props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />;
          })}
        </ul>
      )}
    </div>
  </div>
);

const ConnectedExpenseList = connect((state) => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters),
  };
})(ExpenseList);

export default ConnectedExpenseList;
