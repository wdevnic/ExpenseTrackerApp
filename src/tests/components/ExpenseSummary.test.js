import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseSummary} from '../../component/ExpenseSummary'

test('should correctly render ExpenseSummary with 1 Expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpenseSummary with multiple Expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal={23577}/>)
    expect(wrapper).toMatchSnapshot()
})