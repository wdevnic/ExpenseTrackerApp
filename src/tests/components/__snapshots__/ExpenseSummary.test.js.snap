// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 Expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple Expense 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $235.77
  </h1>
</div>
`;
